### ISSUE TITLE
TailwindCSS 2.0

### ISSUE THREAD

**Issue by MichalBryxi** [josemarluedke/frontile#126]
This might be a long-shot, but I was trying to use frontile together with [TailwindCSS 2.x.x](https://tailwindcss.com/docs/upgrading-to-v2) mostly following [this manual](https://github.com/raytiley/tailwind-ember-example), but it seems like styles are applied only partially. Since there are no big breaking changes in new tailwind I was wondering whether anyone successfully managed to use those two together? 

![Screenshot 2020-12-19 at 16 56 44](https://user-images.githubusercontent.com/847473/102694792-9372e080-4223-11eb-9da9-cc1d45c23373.png)

![Screenshot 2020-12-19 at 16 57 22](https://user-images.githubusercontent.com/847473/102694835-d9c83f80-4223-11eb-8c29-01214dde7327.png)

**Comment by MichalBryxi** [comment]
Linking (maybe) related issue: https://github.com/raytiley/tailwind-ember-example/issues/3

**Comment by josemarluedke** [comment]
Thanks for trying Frontile. I haven't had the time to test with TailwindCSS v2, but I was planning to do so this weekend or next week.

If you have PurgeCSS enabled, make sure to white list the frontile classes: https://frontile.dev/docs/forms#styles

**Comment by MichalBryxi** [comment]
Linked issue has my setup, without the `frontile` specific stuff. I will paste full blown config here as it's IMO relevant:

```js
// ember-cli-build.js

"use strict";

const EmberApp = require("ember-cli/lib/broccoli/ember-app");
const cssImport = require("postcss-import");
const autoprefixer = require("autoprefixer");
const tailwind = require("tailwindcss");

module.exports = function (defaults) {
  let app = new EmberApp(defaults, {
    postcssOptions: {
      compile: {
        plugins: [
          {
            module: cssImport,
          },
          {
            module: autoprefixer,
            options: {},
          },
          {
            module: tailwind,
            options: {
              config: "./config/tailwindcss-config.js",
            },
          },
        ],
      },
    },
    SRI: {
      enabled: false,
    },
    sourcemaps: {
      enabled: true,
      extensions: ["js"],
    },
  });

  return app.toTree();
};
```

```js
// tailwindcss-config.js

module.exports = {
  // TODO: This wipes out bp-ember-components tailwind/frontile styles
  // purge: {
  //   content: ["./app/**/*.html", "./**/*.hbs"],
  //   options: {
  //     safelist: [
  //       /^form-field-checkbox/,
  //       /^form-field-feedback/,
  //       /^form-field-hint/,
  //       /^form-field-input/,
  //       /^form-field-label/,
  //       /^form-field-radio/,
  //       /^form-field-textarea/,
  //       /^form-input/,
  //       /^form-textarea/,
  //       /^form-select/,
  //       /^form-checkbox/,
  //       /^form-radio/,
  //       /^form-checkbox-group/,
  //       /^form-radio-group/,
  //       /^notifications-container/,
  //       /^notification-card/,
  //       /^notification-transition/,
  //       /^btn/,
  //     ],
  //   },
  // },
  darkMode: false, // or 'media' or 'class'
  theme: {
    extend: {
    },
  },
  variants: {
    extend: {
    },
  },
  plugins: [
    require("@frontile/forms/tailwind"),
    require("@frontile/buttons/tailwind"),
    require("@frontile/notifications/tailwind"),
  ],
};
```

**Comment by MichalBryxi** [comment]
OK, I think I have the culprit. The settings used above - even with the uncommented `TODO` section will probably work. The problem is with how tailwind-2 themes [handle `default` styles](https://tailwindcss.com/docs/upgrading-to-v2#update-default-theme-keys-to-default):

```js
// Tailwind v1.x

// tailwind.config.js
module.exports = {
  theme: {
    borderRadius: {
      none: '0',
      sm: '0.125rem',
      default: '0.25rem',
      md: '0.375rem',
      lg: '0.5rem',
    },
  }
}

// Tailwind v2.x

module.exports = {
  theme: {
    borderRadius: {
      none: '0',
      sm: '0.125rem',
      DEFAULT: '0.25rem',
      md: '0.375rem',
      lg: '0.5rem',
    },
  }
}
```

Notice the `default` vs `DEFAULT`. And since frontile seems [to be using default quite a lot](https://github.com/josemarluedke/frontile/blob/v0.8.0/packages/buttons/tailwind/default-options.js#L6) I think this will wipe some important styles.

But maybe I'm wrong with my guess?

**Comment by josemarluedke** [comment]
@MichalBryxi Just published v0.9.0 with TailwindCSS v2.

Please let me know if all the issues you were having are solved. Thank you again for trying Frontile.

**Comment by MichalBryxi** [comment]
I'm using `button`, `form` and `notification` so far and they look ok to me. I just noticed that the color palette seems a little bit different. Left: current frontile readme, right: What I get from frontile 0.9.0.

![Screenshot 2020-12-21 at 19 04 31](https://user-images.githubusercontent.com/847473/102812708-c81f9d00-43c7-11eb-8b5b-925fb69f35d5.png)
 
I don't really understand what the [Tailwind changelog](https://tailwindcss.com/docs/upgrading-to-v2#configure-your-color-palette-explicitly) is saying. I read it as: Default color palette has changed. If you're fine with it, great, if not here is the v1 version.

Personally I don't care that much, but the original colors were less agressive (more pastel-like) to my eyes, so maybe turning down the colors a little bit would be nice?
